src/test/java/hudson/plugins/git/TaskListenerLoggerTest.java

```diff
@@ -0,0 +1,35 @@
+package hudson.plugins.git;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoInteractions;
+
+import hudson.model.TaskListener;
+import org.apache.commons.io.output.NullPrintStream;
+import org.junit.Before;
+import org.junit.Test;
+
+public class TaskListenerLoggerTest {
+
+    private TaskListener listener;
+
+    @Before
+    public void invokeTaskListener() {
+        listener = NullPrintStream::new;
+    }
+
+    @Test
+    public void shouldInvokeTheLoggerMethod() {
+        TaskListenerLogger listenerLoggerMock = mock(TaskListenerLogger.class);
+        listenerLoggerMock.printLogs(listener, true, "This message should be printed.");
+        verify(listenerLoggerMock).printLogs(listener, true, "This message should be printed.");
+    }
+
+    @Test
+    public void shouldNotInvokeTheLoggerMethod() {
+        TaskListenerLogger listenerLoggerMock = mock(TaskListenerLogger.class);
+        listenerLoggerMock.printLogs(listener, false, "This message should be printed.");
+        verify(listenerLoggerMock).printLogs(listener, false, "This message should be printed.");
```

Comment:
One nit was that it would be more understandable for the intent to word the message as "this msg should NOT be printed" ;)

Another aspect is I don't quite understand how this gets tested (that it actually works) but maybe it's just between me and mocking frameworks :) I would have written a test scripted pipeline running int a JenkinsRule, and would inspect its build log with assertions that certain strings are or aren't present. This would also reflect the usage reality as much as possible, although borders on integration rather than unit testing. Still, a common pattern in the Jenkins plugins...

Reply:
Hey @jimklimov 

I have added the `JenkinsRule` test case. I am not sure I am testing the right way. 

My pipeline being build the correct way.